File: basics/28-fibonacci/Solution.java
class Solution {
    public int[] firstFiveFibonacci() {
        int[] result = new int[5];
        int a = 0, b = 1;
        for (int i = 0; i < 5; i++) {
            result[i] = a;
            int next = a + b;
            a = b;
            b = next;
        }
        return result;
    }
}

File: basics/28-fibonacci/Solution.cpp
#include <vector>

std::vector<int> firstFiveFibonacci() {
    std::vector<int> result;
    int a = 0, b = 1;
    for (int i = 0; i < 5; ++i) {
        result.push_back(a);
        int next = a + b;
        a = b;
        b = next;
    }
    return result;
}
